cmd/intervalOverlap/intervalQuery.go

```diff
@@ -38,32 +37,62 @@ func buildTree(intervals []interval.Interval, aggregate bool) map[string]*interv
 	return interval.BuildTree(intervals)
 }
 
-func queryWorker(tree map[string]*interval.IntervalNode, queryChan <-chan interval.Interval, answerChan chan<- *queryAnswer, relationship string, wg *sync.WaitGroup, mergedOutput bool) {
+func queryWorker(tree map[string]*interval.IntervalNode, queryChan <-chan interval.Interval, answerChan chan<- *queryAnswer, relationship string, wg *sync.WaitGroup, mergedOutput bool, thresholdOverlap float64) {
 	var answer []interval.Interval
-	answerTrue := make([]interval.Interval, 1)
 	buf := make([]interval.Interval, 1000)
 	numSeen := 0
 	for query := range queryChan {
 		numSeen++
-		if mergedOutput {
-			answer = interval.Query(tree, query, relationship)
-			answerChan <- &queryAnswer{query, answer}
-			continue
-		}
+		answer = getAnswer(query, tree, relationship, mergedOutput, thresholdOverlap, buf)
+		answerChan <- &queryAnswer{query, answer}
+	}
+	wg.Done()
+}
 
-		// else if no merged output
+func getAnswer(query interval.Interval, tree map[string]*interval.IntervalNode, relationship string, mergedOutput bool, thresholdOverlap float64, buf []interval.Interval) []interval.Interval {
+	var answer []interval.Interval
+
+	// special case to run QueryBool for faster processing
+	if !mergedOutput && thresholdOverlap == 0 {
 		if interval.QueryBool(tree, query, relationship, buf) {
-			answer = answerTrue
+			return make([]interval.Interval, 1)
 		} else {
-			answer = nil
+			return nil
 		}
-		answerChan <- &queryAnswer{query, answer}
 	}
-	wg.Done()
-}
 
-func writeToFile(answerChan <-chan *queryAnswer, outfile io.Writer, mergedOutput bool, nonOverlap bool) {
+	answer = interval.Query(tree, query, relationship)
+	answer = passingThreshold(query, answer, thresholdOverlap)
+
+	if answer == nil {
+		return nil
+	}
+
 	if mergedOutput {
+		return answer
+	}
+
+	// else no merged output
+	return make([]interval.Interval, 1)
+}
+
+func passingThreshold(query interval.Interval, answer []interval.Interval, thresholdOverlap float64) []interval.Interval {
+	if len(answer) == 0 || thresholdOverlap == 0 {
+		return answer
+	}
+	var ovSize, intSize float64
+	for a := range answer {
+		ovSize = float64(interval.OverlapSize(answer[a], query))
+		intSize = float64(interval.IntervalSize(query))
+		if ovSize/intSize >= thresholdOverlap {
+			return make([]interval.Interval, 1)
```

Comment:
I am not sure why we are sending back an slice of intervals with length 1, which I think contains default values for the interval?